### PR TITLE
Groups are not plugged to principal

### PR DESCRIPTION
**Pull request by Natim**
If you create a group and add people to it, then they cannot access records with the group as a permission principal.

### CODE REVIEW

File: kinto/authorization.py [hunk 1]
```diff
@@ -133,6 +133,16 @@ def build_permissions_set(object_uri, unbound_permission,
     return granters
 
 
+# XXX: May need caching
+def groupfinder(userid, request):
+    authn_type = getattr(request, 'authn_type', None)
+    if authn_type is None:
+        return []
+
+    prefixed_userid = '%s:%s' % (authn_type.lower(), userid)
```
Comment (leplatrem) [inline kinto/authorization.py]: Now we have ``request.prefixed_userid`` :)

File: kinto/views/groups.py [hunk 2]
```diff
@@ -61,7 +64,11 @@ def process_record(self, new, old=None):
         if old is None:
             existing_record_members = set([])
         else:
-            existing_record_members = set(old['members'])
+            try:
+                existing_record_members = set(old['members'])
+            except KeyError:
```
Comment (leplatrem) [inline kinto/views/groups.py]: A test for this edge-case ?

Comment (Natim) [inline kinto/views/groups.py]: sure.

File: kinto/authorization.py (hunk 1 above)
Comment (Natim) [inline kinto/authorization.py]: I tried but it seams that it is not populated yet at this time.

Comment (Natim) [inline kinto/authorization.py]:
```
  File "/home/rhubscher/mozilla/kinto/kinto/authorization.py", line 139, in groupfinder
    request.prefixed_userid)
AttributeError: 'Request' object has no attribute 'prefixed_userid'
```

File: kinto/views/groups.py [hunk 3]
```diff
@@ -64,7 +64,10 @@ def process_record(self, new, old=None):
...
+                existing_record_members = set([])
```
Comment (leplatrem) [inline kinto/views/groups.py]: existing_record_members = set(old.get('members', []))

File: kinto/tests/test_views_groups.py [hunk 4]
```diff
@@ -58,6 +58,20 @@ def test_wrong_create_permissions_cannot_be_added_on_groups(self):
                           headers=self.headers,
                           status=400)
 
+    def test_recreate_group_after_deletion_works(self):
```
Comment (leplatrem) [inline kinto/tests/test_views_groups.py]: works ?
